what4/src/What4/BaseTypes.hs

```diff
@@ -289,13 +292,20 @@ instance Show (BaseTypeRepr bt) where
 instance ShowF BaseTypeRepr
 
 instance Pretty (FloatPrecisionRepr fpp) where
-  pretty = viaShow
+  pretty :: FloatPrecisionRepr fpp -> Doc ann
+  pretty (FloatingPointPrecisionRepr exp' sig) = 
+    cat ["FloatingPrecision-", pretty $ natValue exp' , "-", pretty $ natValue sig]
```

Comment:
For consistency with how `crucible`'s [`FloatRepr` is pretty-printed](https://github.com/GaloisInc/crucible/blob/21935341bcb8f375b0c1d62247c6f6bfc2390fb6/crucible/src/Lang/Crucible/Types.hs#L516), I'd prefer to separate the fields of `FloatingPointPrecisionRepr` with spaces instead of hyphens. Something like this:

```suggestion
    parens ("FloatingPrecision" <+> pretty (natValue pretty) exp' <+> pretty (natValue sig))
```

(I also added a call to `parens`, again for consistently with `crucible`'s output.)